indico/modules/events/papers/controllers/display.py

```diff
@@ -177,3 +187,30 @@ def _process(self):
         return WPDisplayReviewingArea.render_template('display/reviewing_area.html', self._conf, event=self.event_new,
                                                       contribs_to_review=contribs_to_review,
                                                       reviewed_contribs=reviewed_contribs)
+
+
+class RHJudgePaper(RHPaperBase):
+    def _check_paper_protection(self):
+        return self.paper.can_judge(session.user, check_state=True)
+
+    def _process(self):
+        form = PaperJudgmentForm()
+        if form.validate_on_submit():
+            judge_paper(self.paper, form.judgment.data, form.judgment_comment.data, judge=session.user,
+                        send_notifications=form.send_notifications.data)
+            return jsonify_data(flash=False, html=render_paper_page(self.paper))
+
+
+class RHResetPaperState(RHPaperBase):
+    def _check_paper_protection(self):
+        if self.paper.state == PaperRevisionState.submitted:
+            return False
+        # managers and judges can always reset
+        if self.paper.event_new.can_manage(session.user) or self.paper.can_judge(session.user):
+            return True
+
+    def _process(self):
+        if self.paper.state != PaperRevisionState.submitted:
+            reset_paper_state(self.paper)
+            flash(_("Paper state has been reset"), 'success')
```

Comment:
Wouldn't "Judgment has been reset" be much more suitable here (same in the equivalent abstracts RH btw)?